app/views/admin/navigation_links/_form.html.erb

```diff
@@ -8,10 +8,21 @@
   <%= form.text_field :url, class: "form-control" %>
   <div class="alert alert-info">A full (external) or relative (internal) URL for the link</div>
 </div>
-<div class="form-group">
-  <%= form.label :icon %>
-  <%= form.text_area :icon, class: "form-control" %>
-  <div class="alert alert-info">An SVG icon should be pasted in the textarea</div>
+<div class="form-group" data-controller="svg-icon-upload">
+  <div nav-link-id="<%= form.object[:id] %>">
+    <%= form.label :icon %>
+    <div id="svg-icon-preview" class="<%= form.object[:id].nil? ? "" : "pb-3" %>" style="width: 24px;">
```

Comment:
If it's posible, can we move the width of the preview in the CSS? BTW 24px is really small, I'd double it to 48px